### PR TITLE
Typechecking in vdom and task

### PR DESCRIPTION
**Pull request by pbiggar**
This adds all type signatures to vdom, and some to task.

I also removed commented out code from vdom (where it didn't look like it was intended to come back).

### CODE REVIEW

File: src-ocaml/vdom.ml [hunk 1]
```diff
@@ -33,79 +27,87 @@ type 'msg property =
...
-  (* | ArrayNode of string * string * string * string * 'msg property array * 'msg t array *)
   (* LazyGen (key, fnGenerator) *)
   | LazyGen of string * (unit -> 'msg t) * 'msg t ref
-  (* Tagger (toString, toDom, toVNodes) *)
-(* | Tagger of (unit -> string) * ('msg applicationCallbacks ref -> Web.Node.t -> Web.Node.t -> int ->  'msg t list -> Web.Node.t) * (unit -> 'msg t) *)
   (* Tagger (tagger, vdom) *)
   | Tagger of ('msg applicationCallbacks ref -> 'msg applicationCallbacks ref) * 'msg t
-  (*  *)
-  (* | Tagger of (('a -> 'msg) -> 'a t -> 'msg t) *)
-  (* Custom (key, cbAdd, cbRemove, cbChange, properties, children) *)
```
Comment (OvermindDL1) [inline src-ocaml/vdom.ml]: Technically this one is going to come back once I get around to adding support for it in, can you keep it in so it acts as a reminder for me, maybe add a `TODO` comment for it for easily greppability?  :-)

**Comment by OvermindDL1** [comment]
Also, the javascript files are interestingly out of date with your PR, can you run `npm run pretest` just to make sure they get updated too?

File: src-ocaml/vdom.ml (hunk 1 above)
Comment (OvermindDL1) [reply src-ocaml/vdom.ml]: For note, I'm speaking of the `Custom` node, not anything else.  :-)

**Comment by pbiggar** [comment]
@OvermindDL1 Done!

**Comment by OvermindDL1** [comment]
That was fast!  ^.^

Awesome, looks great!!

Merging in, no point on a new NPM release yet since it won't effect its runnability or IDE usages.  :-)

Thanks much!!!

**Comment by IanConnolly** [comment]
@OvermindDL1 this seems to have broken the API for Tasks. The concretizing of various tasks to `('value, string)` seems undesirable to me. It seems all `('value, string)` instances should be `('succeed, 'fail)` (ie. variadic in the type of the failure case). Is that correct? I can fix if so.

**Comment by OvermindDL1** [comment]
> We caught this specific error after we pointed our bucklescript-tea dependency in our app to master, so we'll catch this sort of mistake faster in future.

Awesome, it was great to catch before the release!  :-)

**Pull request merged**